### PR TITLE
Create tabular_parser

### PR DESCRIPTION
**Pull request by vguillet**
Add support for reading CSV data

### CODE REVIEW

**Comment by vguillet** [comment]
The reader still needs to be added to base.py, I'll take care of doing it if this reader is acceptable

**Comment by vguillet** [comment]
Debated adding excel reader, but first want to know if this CSV reader is acceptable

File: gpt_index/readers/file/tabular_parser [hunk 1]
```diff
@@ -0,0 +1,30 @@
+"""Tabular parser.
```
Comment (jerryjliu) [inline gpt_index/readers/file/tabular_parser:1]: can you make this have a .py extension?

File: gpt_index/readers/file/tabular_parser [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        try:
+            import csv
+        except ImportError:
+            raise ValueError("csv module is required to read CSV files.")
+        text_list = []
+        with open(file, "r") as fp:
+            csv_reader = csv.reader(fp)
+            for row in csv_reader:
+                text_list.append(", ".join(row))
+        text = ";\n".join(text_list)
```
Comment (jerryjliu) [inline gpt_index/readers/file/tabular_parser:28]: why do we need the semicolon?

**Review by jerryjliu — COMMENTED** [review]
thanks for doing this! a few comments. also: 
1. can you test on a .csv dataset? 
2. when you do test on a .csv dataset, can you post a screenshot? (will include in the twitter release :) )

File: gpt_index/readers/file/tabular_parser (hunk 1 above)
Comment (vguillet) [reply gpt_index/readers/file/tabular_parser:1]: woops, yep on it

File: gpt_index/readers/file/tabular_parser (hunk 2 above)
Comment (vguillet) [reply gpt_index/readers/file/tabular_parser:28]: That is not strictly necessary indeed, \n is enough, ill adjust that

File: gpt_index/readers/file/tabular_parser (hunk 1 above)
Comment (vguillet) [reply gpt_index/readers/file/tabular_parser:1]: Also if I understand correctly, you want a sceenshot of a csv parsed to text?

Comment (vguillet) [reply gpt_index/readers/file/tabular_parser]: image

**Review by jerryjliu — COMMENTED** [review]
looks good, i can help with some of the linting

**Review by jerryjliu — APPROVED** [review]

**Pull request merged**